packages/playwright/tests/fixtures-app/app/_pw-utils/network-mock-global-setup.ts

```diff
@@ -0,0 +1,121 @@
+import type { FullConfig } from "@playwright/test";
+import type {
+  addOriginalSourceResult,
+  addSourceMapResult,
+  beginRecordingUploadParameters,
+  beginRecordingUploadResult,
+  endRecordingUploadResult,
+  existsResult,
+  setAccessTokenResult,
+  setRecordingMetadataResult,
+} from "@replayio/protocol";
+import { WebSocket } from "undici";
+
+async function globalSetup(_config: FullConfig) {
```

Comment:
I think it will be best to pass a function through this config, any given test could pass a testing function in `.use` within its own `playwright.config.mts` and we could call this function from here with the appropriate context object. 

This way each test could keep its test and mocking body close to itself (colocation, yay!). 

This might require us to stringify the said function but I think it's fine. It's a standard constraint when dealing with `page.evaluate` and similar APIs as those pass "functions" (in a form of string) across runtime boundaries